Removing additional blockflattener pass

polgreen opened the pull request:
This block flattener pass doesn't seem necessary. If it is necessary, we should add some tests that fail when it is removed

pramodsu commented:
I don't think any of the block flatteners are strictly necessary -- they just flatten unnecessarily nested code like:

```
{
   s1;
   {
      s2;
   }
}
```

Into:

```
{
   s1;
   s2;
}
````

But everything in uclid5 is supposed to work even if don't do any flattening. The purpose of adding the flattening pass was to try and make the resulting output slightly easier to read.

polgreen commented:
But does it change the behaviour if it is run twice vs once? The original order of passes is this, and I've removed the first block flattener:
~~~
 passManager.addPass(new BlockFlattener()) // I've removed this block flattener pass
 passManager.addPass(new NewInternalProcedureInliner())
 passManager.addPass(new PrimedVariableCollector())
 passManager.addPass(new PrimedVariableEliminator())
 passManager.addPass(new PrimedHistoryRewriter())
 passManager.addPass(new IntroduceFreshHavocs())
 passManager.addPass(new RewriteFreshLiterals())
 passManager.addPass(new BlockFlattener())
~~~
